bot.py

```diff
@@ -127,12 +148,19 @@ async def add(ctx):
             }
         )
     )
-    add_to = await bot.wait_for("message", timeout=60)
-    if add_to:
-        tracker.add_inv(author, add_to.content)
-
-
-@bot.command()
+    while True:
+        try:
+            add_message = await bot.wait_for("message", timeout=60)   
+            if ctx.message.channel.id == add_message.channel.id and author == add_message.author.id:
+                tracker.add_inv(author, add_message.content)
+                await add_message.add_reaction("✅")
+                break
+        except:
+            await ctx.message.channel.send(f"Inventory add request timed out. Please use {bot_prefix}inv add again.")
+            break
```

Comment:
Since [`bot.wait_for()`](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html?highlight=command%20ext%20bot#discord.ext.commands.Bot.wait_for) just raises an [`asyncio.TimeoutError`](https://docs.python.org/3/library/asyncio-exceptions.html#asyncio.TimeoutError) when the timeout occurs, we can just catch that. The suggestion below is dependent on adding the following import: `from asyncio import TimeoutError`

```suggestion
    while True:
        try:
            add_message = await bot.wait_for("message", timeout=60)
        except TimeoutError:
            await ctx.message.channel.send(
                f"Inventory add request timed out. Please use `{bot_prefix}inv` add again."
            )
            break

        if (
            ctx.message.channel.id == add_message.channel.id
            and author == add_message.author.id
        ):
            tracker.add_inv(author, add_message.content)
            await add_message.add_reaction("✅")
            break
```

I have the same thoughts about `author` in this case as in my prior comment. I believe it should be `author_id` if it is the author's ID. Though, this is outside of the changes in this function and should be refactored later.

Reply:
Suggested changes made. Thanks!